### PR TITLE
Hotfix/reporting sections

### PR DESCRIPTION
**Pull request by Sup3rGeo**
Adresses https://github.com/astraw38/pytest-assume/issues/48

### CODE REVIEW

**Comment by Sup3rGeo** [comment]
@astraw38 opened anyways with the double hook because at least we have a functional test case for the issue.

**Comment by Sup3rGeo** [comment]
It actually works with `hookwrapper=True, tryfirst=True`, although I am not 100% certain of other implications

File: tests/test_expect.py [hunk 1]
```diff
@@ -590,3 +590,48 @@ def test_bar(self):
...
+            yield
+            logging.info("Teardown")
+
+        def test_fail_assume(setup_teardown):
+            logging.info("This is a test log")
+            with assume: assert 1 == 2
+        """
+    )
+
+    testdir.makeconftest(
```
Comment (astraw38) [inline tests/test_expect.py:614]: Minor test improvements:


Drop the makereport from the conftest, add:

```py
logging.getLogger().setLevel(logging.DEBUG)
```

to the top of the 'test file'.

Then assert that we have Captured log setup, Captured log call, and Captured log teardown in `"\n".join(result.outlines)`.

**Comment by astraw38** [comment]
I know tryfirst can impact other plugins. I've tried not to touch that, since i know it can prevent downstream plugins from acting on the outcome.

File: tests/test_expect.py (hunk 1 above)
Comment (Sup3rGeo) [reply tests/test_expect.py:614]: The thing is that the bug does not prevent the logging to be present in the output, only prevents it from appearing in the result object returned by the hook calls. Hence the usage of of the conftest so we can analyse the reporting object from inside the hook itself.

Comment (astraw38) [reply tests/test_expect.py:614]: Hmm, that's not what I saw when i created a test to verify the behavior. Commenting out the `pytest_pyfunc_call` hook (only using the `pytest_runtest_call` hook), then running the following test fails locally:

```py

def test_report_sections(testdir):
    testdir.makepyfile(
        """
        import pytest
        import logging
        from pytest import assume
        logging.getLogger().setLevel(logging.DEBUG)
        
        @pytest.fixture
        def setup_teardown():
            logging.info("Setup")
            yield
            logging.info("Teardown")

        def test_fail_assume(setup_teardown):
            logging.info("This is a test log")
            with assume: assert 1 == 2
        """
    )
    result = testdir.runpytest_inprocess()
    result.assert_outcomes(failed=1)
    assert "Captured log call" in "\n".join(result.outlines)
```

**Comment by Sup3rGeo** [comment]
I believe that this shouldn't be a problem with hookwrappers (only with non-hookwrappers hooks returning something where `firstresult=True`, which also does not seem to be the case for `pytest_runtest_call`)

File: tests/test_expect.py (hunk 1 above)
Comment (Sup3rGeo) [reply tests/test_expect.py:614]: you are right and I know where I got confused - the live logs are actually unaffected (the ones I am used to see), only the captured ones that are shown in the report object and in the console together with the exception traceback.

I will update the test to be like this

**Comment by Sup3rGeo** [comment]
@astraw38 I am personally okay/happy with this I guess

**Pull request merged**